services/QuillLMS/client/app/bundles/PremiumHub/containers/__tests__/overviewSection.test.tsx

```diff
@@ -37,23 +100,71 @@ describe('OverviewSection', () => {
+      expect(screen.getByRole('columnheader', { name: /average activities & time spent/i })).toBeInTheDocument()
+      expect(screen.getByRole('columnheader', { name: /post diagnostic completed/i })).toBeInTheDocument()
+      expect(screen.getByRole('columnheader', { name: /overall skill growth/i })).toBeInTheDocument()
+      expect(screen.getByText(/there is no diagnostic data available for the filters selected\. try modifying or removing a filter to see results\./i)).toBeInTheDocument()
+    })
+    test('it should render the expected data when data is present', () => {
+      props.passedData = mockData
+      const { asFragment } = render(<OverviewSection {...props} />);
+      expect(asFragment()).toMatchSnapshot();
+      expect(screen.getByRole('cell', { name: /starter baseline diagnostic \(pre\)/i })).toBeInTheDocument()
+      expect(screen.getByRole('cell', { name: /57 of 65 Students/i })).toBeInTheDocument()
+      expect(screen.getByRole('cell', { name: /58 Students/i })).toBeInTheDocument()
+      expect(screen.getByRole('cell', { name: /17 activities \(26:35\)/i })).toBeInTheDocument()
+      expect(screen.getByRole('cell', { name: /27 of 65 students/i })).toBeInTheDocument()
+      expect(screen.getByRole('button', { name: /\+20%/i})).toBeInTheDocument()
+
+      expect(screen.getByRole('cell', { name: /intermediate baseline diagnostic \(pre\)/i })).toBeInTheDocument()
+      expect(screen.getByRole('cell', { name: /40 of 55 Students/i })).toBeInTheDocument()
+      expect(screen.getByRole('cell', { name: /32 Students/i })).toBeInTheDocument()
+      expect(screen.getByRole('cell', { name: /10 activities \(16:35\)/i })).toBeInTheDocument()
+      expect(screen.getByRole('cell', { name: /25 of 55 students/i })).toBeInTheDocument()
+      expect(screen.getByRole('button', { name: /\+50%/i })).toBeInTheDocument()
+
+      expect(screen.getByRole('cell', { name: /grade 7/i })).toBeInTheDocument()
+      expect(screen.getByRole('cell', { name: /27 of 35 Students/i })).toBeInTheDocument()
+      expect(screen.getByRole('cell', { name: /50 Students/i })).toBeInTheDocument()
+      expect(screen.getByRole('cell', { name: /12 activities \(26:35\)/i })).toBeInTheDocument()
+      expect(screen.getByRole('cell', { name: /17 of 35 students/i })).toBeInTheDocument()
+      expect(screen.getByRole('button', { name: /\+10%/i })).toBeInTheDocument()
+
+      expect(screen.getByRole('cell', { name: /grade 9/i })).toBeInTheDocument()
+      expect(screen.getByRole('cell', { name: /30 of 30 Students/i })).toBeInTheDocument()
+      expect(screen.getByRole('cell', { name: /9 Students/i })).toBeInTheDocument()
+      expect(screen.getByRole('cell', { name: /23 activities \(26:35\)/i })).toBeInTheDocument()
+      expect(screen.getByRole('cell', { name: /10 of 30 students/i })).toBeInTheDocument()
+      expect(screen.getByRole('button', { name: /\+30%/i })).toBeInTheDocument()
+    })
```

Comment:
I appreciate the thoroughness of the tests both here and in the following file, but it kind of feels to me like a snapshot might actually be more useful here --if something were to break here, I think it would be hard to figure out what had happened (as opposed to a snapshot when you can actually see the whole HTML structure). I don't think we need to remove what you have here, but maybe add snapshots too? Also, spacing between the tests would help with readability.

Reply:
that's a good point-- I'll add snapshots and add some spacing to break up the tests